internal/hcl/attribute.go

```diff
@@ -594,16 +594,51 @@ func mockExpressionCalls(expr hcl.Expression, diagnostics hcl.Diagnostics, mocke
 			MarkerRange: t.MarkerRange,
 		}
 	case *hclsyntax.BinaryOpExpr:
+		// Logical operators (|| and &&) expect bools on both sides, so we need to mock the expression
+		// with a bool value.
+		impl := t.Op.Impl
+		params := impl.Params()
+		lhsParam := params[0]
+		rhsParam := params[1]
```

Comment:
is this always expected to be a slice of this size?

Reply:
we should make add a size check just in case?

Reply:
Yeah, all BinaryOp functions have 2 params, but updated this anyway to make sure.